### DISCUSSION TITLE
Monitor Client Connection Pool

### DISCUSSION THREAD

**Discussion by agaengel** [resteasy/resteasy#3525] [Q&A]
Is it possible to monitor the connection pool of a resteasy client?

**Comment by jamezp** [comment]
What do you mean by monitor it? What container are you running in?

**Comment by agaengel** [comment]
We use resteasy with spring. Other pools like org.apache.commons.dbcp2 or org.apache.commons.pool2 are monitored via jmx-exporter and prometheus.

Is there some way to expose the stats of the PoolingHttpClientConnectionManager ?

**Reply by jamezp** [reply]
Ah, the `PoolingHttpClientConnectionManager` is part of the Apache HTTP Client and is only used pool size on the builder has been set. By default it's set to 50 so that connection manager is used. To my knowledge there isn't really a way to monitor those connections.

One possible option would be to create your own `ClientHttpEngineBuilder`, but that feels a bit complicated.

**Reply by agaengel** [reply]
would it be possible to extend the ClientHttpEngine to allow access to a connectionManager if configured ?

**Reply by jamezp** [reply]
It doesn't look like it. It's set in the builder and it doesn't look like there is a way to set it.
https://github.com/resteasy/resteasy/blob/58307d89dded6192fbe1b7b35b9527af07ea3c62/resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ClientHttpEngineBuilder43.java#L160-L173

**Reply by agaengel** [reply]
yes the current version does not provide access to this but would it be a good idea to extend ClientHttpEngine to provide this in general.

I this is a general interest for resteasy I could try to create a PR for this

**Reply by jamezp** [reply]
It's possible we could do something where we attempt to look it up as a provider or in the configuration of the client, but we can't really add a new method as it's implementation specific. The Apache HTTP Client will likely not be the default client in future releases.

**Reply by agaengel** [reply]
hi @jamezp what is the plan for the next default client?

will this client support pooling?

**Reply by jamezp** [reply]
Hi @agaengel, there is no definite plan as of yet. I've experimented with using the Java HTTP Client. There are some issues using it as the Jakarta REST Client Builder allows a `HostnameVerifier` to be set which is not supported on the client.

We may end up using that or a completely different client. I do need to make a plan for this as I feel HTTP/2 support is critical.
